### DISCUSSION TITLE
autodl irssi

### DISCUSSION THREAD

**Discussion by K4Li** [ryelogheat/xpbot#35] [General]
To automate the upload script it requires autodl irssi. Could the same thing be accomplished with autobrr? https://github.com/autobrr/autobrr

**Comment by ryelogheat** [comment]
Probably, I don't have any personal experience with autobrr but I did see they mention they support _custom commands_ so you could theoretically run auto_upload.py & its required args after you get a match 

Do you know if autobrr monitors the status of downloads? Like after it gets a match & sends the torrent to your client, does it keep track of the torrents download/progress? 

The reason why that's necessary ^^ is so you can run auto_upload.py only when the download actually completes, the way I do this with autodl is through labels, editing rtorrent.rc, and having rtorrent itself trigger the upload script when the download completes (basically a specific process that only applies to autodl & rtorrent)

**Comment by ryelogheat** [comment]
If autobrr keeps track of the torrents download progress & can execute a script when it completes, then this should be quite simple to implement. If it doesn't track a torrents download progress then your going to need to find a way make your download client trigger auto_upload.py when the download completes 

btw here is the rtorrent.rc line that I used to make that ^^ happen 
`method.set_key = event.download.finished,complete,"execute=/PATH/TO/rtorrent_on_complete_reupload.sh,$d.name=,$d.data_path=,$d.custom1=,$d.custom=upload_to_tracker"`
in that bash script I configured it so that it won't re-upload unless the 4th arg (upload_to_tracker) is present. That's how I excluded regular radarr/sonarr downloads from re-uploading

**Comment by ryelogheat** [comment]
Actually while I got you here, can you email me at ryelogheat@pm.me I wanna talk to you about something but I wasn't able to find your email address on your github profile
